editor/components/provider/index.js

```diff
@@ -49,6 +50,15 @@ class EditorProvider extends Component {
 			...props.settings,
 		};
 
+		// Provide Backwards compatibility for enabling wide image support with `wide-images`.
+		if ( ! this.settings.alignWide && this.settings.wideImages === true ) {
+			this.settings.alignWide = true;
+			console.warn( __( "Adding theme support for `wide-images` inside the `gutenberg` array is deprecated.\n" +
+				"Instead, use `add_theme_support( 'align-wide' );`.\n" +
+				"See https://wordpress.org/gutenberg/handbook/reference/theme-support/"
+			) );
+		}
+
```

Comment:
Not sure if I'm doing this in the right spot. Feel free to point me in another direction if not.

Reply:
RE Travis errors:
Looks like I need to use a different quote style here, which is easy enough (whoops). That won't fix that it doesn't like that I'm outputting to the console, so going to dig a bit more on the other warning you referenced @youknowriad.

Reply:
Looks like in the other case (https://github.com/WordPress/gutenberg/blob/master/blocks/hooks/matchers.js#L12), the console warning is being ignored through a comment.

Pushed another commit that should help ESLint / Travis pass.

Reply:
Yep, it helps if you can install an `eslint` plugin in your favorite IDE to catch the errors "live". You can also run `npm run lint` locally.

---
That said, I wonder if this check should be done server-side instead,

```php
if (! empty( $gutenberg_theme_support)) {
wp_add_inline_script(
	'wp-editor',
	'console.warn("message")'
);
}
```

and update the setting

```php
$editor_settings = array(
  'alignWide' => $align_wide || ! empty( $gutenberg_theme_support[0]['wide-images'] )
)
```

That way the client is consistent. What do you think?

Reply:
Thanks! For some reason, I (incorrectly) assumed that lint was run as part of build when using `npm run dev`.  Have been running it manually to check since learning otherwise, because some of the editing is in vim, and I don't have an `eslint` plugin installed there yet.

I'd been assuming the client/JS side due to the `console.warn()` suggestion, but have no problem at all with that route.

If we're going to solve it server-side, should it be with `_doing_it_wrong` like @aaronjorbin suggested, or is sending the warning across via `wp_add_inline_script()` more idiomatic and/or likely to be noticed by devs?

Reply:
> If we're going to solve it server-side, should it be with _doing_it_wrong like @aaronjorbin suggested, or is sending the warning across via wp_add_inline_script() more idiomatic and/or likely to be noticed by devs?

Not strong opinion myself. I'd personally use `console.warn` for consistency with the other "depreacted" API we have in Gutenberg. I don't know if plugins/theme authors are already familiar with the `_doing_it_wrong` behavior in other parts of WordPress, in which case, it would be a better choice.

In other words, I'm personally ok with both. The important thing is that the user notices it. What do you think @aaronjorbin ?

Reply:
I'll go ahead and put something together, and we can always swap out the method upon feedback.

Another question: Any suggestions on the best naming for the `colors` option, since I'm assuming we probably don't want to go with `add_theme_support( 'colors' );`?

Reply:
`editor-color-palette` perhaps?

Reply:
> `editor-color-palette`

That sounds good.